assets/js/theme/common/carousel/index.js

```diff
@@ -0,0 +1,73 @@
+import 'slick-carousel';
+
+import {
+    dotsSetup,
+    setTabindexes,
+    arrowAriaLabling,
+    heroCarouselSetup,
+    getRealSlidesQuantityAndCurrentSlide,
+} from './utils';
+
+const onCarouselChange = (event, carousel) => {
+    const {
+        options: { prevArrow, nextArrow },
+        currentSlide,
+        slideCount,
+        $dots,
+        $slider,
+        breakpointSettings,
+        activeBreakpoint,
+    } = carousel;
+
+    const { realSlideCount, realSlide } = getRealSlidesQuantityAndCurrentSlide(
+        breakpointSettings,
+        activeBreakpoint,
+        currentSlide,
+        slideCount,
+        $slider.data('slick').slidesToScroll,
+    );
+
+    const $prevArrow = $slider.find(prevArrow);
+    const $nextArrow = $slider.find(nextArrow);
+
+    dotsSetup($dots, realSlide, realSlideCount, $slider.data('dots-labels'));
+    setTabindexes($slider.find('.slick-slide'), $prevArrow, $nextArrow, realSlide, realSlideCount);
+    arrowAriaLabling($prevArrow, $nextArrow, realSlide, realSlideCount);
+};
+
+export default function () {
+    const $carouselCollection = $('[data-slick]');
+
+    if ($carouselCollection.length === 0) return;
+
+    $carouselCollection.each((index, carousel) => {
+        // getting element using find to pass jest test
+        const $carousel = $(document).find(carousel);
+
+        if ($carousel.hasClass('productView-thumbnails')) {
+            $carousel.slick();
+            return;
+        }
+
+        $carousel.on('init', onCarouselChange);
+        $carousel.on('afterChange', onCarouselChange);
+
+        const isMultipleSlides = $carousel.children().length > 1;
+        const customPaging = isMultipleSlides
+            ? () => (
+                '<button type="button"></button>'
+            )
+            : () => {};
+
+        const options = {
+            accessibility: false,
+            arrows: isMultipleSlides,
+            customPaging,
+            dots: isMultipleSlides,
+        };
```

Comment:
since we don't reuse this const and this won't affect readability don't use think to move the options object to carousel.slick call without const?

Reply:
make sence